lib/vdsm/storage/sdm/api/copy_data.py

```diff
@@ -146,15 +141,17 @@ class CopyDataDivEndpoint(properties.Owner):
                                     maxval=sc.MAX_GENERATION)
     prepared = properties.Boolean(default=False)
 
-    def __init__(self, params, host_id, writable, is_destination=False,
-                 lock_image=True):
+    def __init__(self, params, host_id, writable, dest=None):
         self.sd_id = params.get('sd_id')
         self.img_id = params.get('img_id')
         self.vol_id = params.get('vol_id')
         self.generation = params.get('generation')
         self.prepared = params.get('prepared')
-        self.is_destination = is_destination
-        self.lock_image = lock_image
+        self.is_destination = dest is None
+        self.lock_image = (
+            self.is_destination
+            or dest.get('img_id') != self.img_id
+        )
```

Comment:
The check is confusing but correct - the destination always lock the image, the source (dest != None) lock the image only if dest and source have different image id. Maybe worth a comment to explain this.

Need to update to use an endpoint object instead of dict

    getattr(dest, "img_id", None) != self.img_id

Or add img_id property to external endpoint - it can return vol_id or None since
external endpoint does not have an image in all cases. It is easier to have the
same interface instead of using getattr().